casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -94,8 +93,35 @@ object BlockCreator {
         // parent in 1 situation. It would be safer for the validator to make sure all parents post state
         // are in active state.
         isActive = parents
-          .traverse(b => isActiveValidator(b, runtimeManager, validatorIdentity))
+          .traverse(b => isActiveValidator(b, runtimeManager, validatorIdentity.publicKey))
           .map(_.forall(identity))
+
+        // evil validator can be slashed in previous block
+        // the evil should be slashed when it is active to avoid unnecessary double slashing.
+        invalidLatestMessageExcludeInactive <- invalidLatestMessages.toList.filterA(
+                                                (v) =>
+                                                  parents
+                                                    .traverse(
+                                                      b =>
+                                                        isActiveValidator(
+                                                          b,
+                                                          runtimeManager,
+                                                          PublicKey(v._1)
+                                                        )
+                                                    )
+                                                    .map(_.forall(identity))
+                                              )
```

Comment:
Slashed validator is not removed from active validator set until next epoch. We should check bonds list if validator exists on the list and has stake greater that zero. I'm not sure is it necessary to check if it's active but it makes sense to have this check also.

I don't like how `isActiveValidator` is called for each parent and with this new check we will use it in even more cases. Each call is doing Rholang execution and it would better is this function can accept list of keys and return results for all of them in just one Rholang execution. The same should be for calling bonds list.

Reply:
Ok. You are right. I misunderstand in the first place.